buildSrc/build.gradle.kts

```diff
@@ -3,9 +3,10 @@ plugins {
 }
 
 dependencies {
+    implementation(files(libs.javaClass.superclass.protectionDomain.codeSource.location))
```

Comment:
is this needed?

Reply:
For now, Is required to access version catalog inside convention plugin. 
It seems that better support for version catalog will be release in future versions of Gradle.

Reply:
Context: https://github.com/gradle/gradle/issues/15383#issuecomment-779893192

According to a reply in this issue, Gradle 8.5 has some [new support](https://docs.gradle.org/8.5/release-notes.html#catalog-precompiled) for this so you could probably consider using that.

Reply:
@AntsyLich @stevenyomi Yeah I tried that, It looks worse to me. 
You can't access the variable directly, you need to find by string, which is prone to error. 

```kt
dependencies {
    versionCatalogs
        .named("libs")
        .findLibrary("detekt-rules-formatting")
        .ifPresent {
            detektPlugins(it)
        }

    versionCatalogs
        .named("libs")
        .findLibrary("detekt-rules-compose")
        .ifPresent {
            detektPlugins(it)
        }
}
```

It's up to you guys. I think using the import, for now, its a better solution.

Reply:
Should I make the change? What's the decision here?

Reply:
I think this is acceptable:

```kotlin
dependencies {
    val libs = versionCatalogs.named("libs")
    detektPlugins(libs.findLibrary("detekt.rules.formatting").get())
    detektPlugins(libs.findLibrary("detekt.rules.compose").get())
}
```

I'm just providing context, and won't review or decide anything.

Reply:
@stevenyomi Good catch, I missed the `.get()` method. Your solution works and is better than using `ifPresent`... I still prefer importing the `libs class` and accessing the declarations directly though, it is safer and we don't rely on strings. Anyway, I'm good with both solutions.

@AntsyLich Let me know what you think.

Reply:
Keep it as it is. I'll do some refactoring before merging

Reply:
What kind of refactoring? I'm open to requests.
I'm in a team transition period at work, so I have free time lol.

Btw, my idea here is to open another PR with Detekt auto-fixes after merging this one.

Reply:
Gonna make something like https://github.com/android/nowinandroid/tree/main/build-logic. I've WIP and will merge this after that.

Reply:
FYI the `build-logic` you linked is the same as the `buildSrc` directory. Now in Android makes use of binary plugins, and the thing we're talking about is for precompiled script plugins. Reference: https://proandroiddev.com/sharing-build-logic-with-kotlin-dsl-203274f73013#ac6a

Reply:
I'm aware. Gonna move the scripts inside `build-logic`

Reply:
I haven't dug very deep but it seems Gradle 8.x already had optimizations on `buildSrc`, so you'd probably need to research further what are the current drawbacks of `buildSrc`.

https://github.com/gradle/gradle/pull/22540
https://docs.gradle.org/8.0/release-notes.html#improvements-for-buildsrc-builds
https://github.com/android/nowinandroid/discussions/962

Reply:
@AntsyLich Good idea. I think it will be a good improvment.
I usually structure my projects like this:
https://github.com/theolm/kmm-template/tree/main/build-logic

Do you have something ready yet? otherwise I can take this refactor.